lib/evmone/baseline.cpp

```diff
@@ -51,12 +51,19 @@ template <evmc_opcode Op>
 inline evmc_status_code check_requirements(
     const CostTable& cost_table, ExecutionState& state) noexcept
 {
-    if (INTX_UNLIKELY(cost_table[Op] == instr::undefined))
-        return EVMC_UNDEFINED_INSTRUCTION;
+    auto gas_cost = instr::gas_costs[EVMC_FRONTIER][Op];  // Init assuming const cost.
+    if constexpr (!instr::has_const_gas_cost(Op))
+    {
+        gas_cost = cost_table[Op];  // If not, load the cost from the table.
 
-    if (INTX_UNLIKELY((state.gas_left -= cost_table[Op]) < 0))
-        return EVMC_OUT_OF_GAS;
+        // Negative cost marks an undefined instruction.
+        // This check must be first to produce correct error code.
+        if (INTX_UNLIKELY(gas_cost < 0))
```

Comment:
Does the change from `== instr::undefined` improves performance?

Reply:
As it's now done only for non-const-gas instructions, this function will not produce correct result for always undefined instructions, so add something like
```
static_assert(instr::gas_costs[EVMC_FRONTIER][Op] != instr::undefined || !instr::has_const_gas_cost(Op));
```

Reply:
> Does the change from `== instr::undefined` improves performance?

Not really. It saves one byte of the code.